core/core.go

```diff
@@ -108,6 +111,18 @@ func New(config Config) (*App, error) {
 	log.SetLevel(log.Level(config.Verbosity))
 	log.AddHook(loghooks.NewKeySuffixHook())
 
+	// Load private key and add peer ID hook.
+	privKeyPath := filepath.Join(config.DataDir, "keys", "privkey")
```

Comment:
As a side effect of not being able to call `p2p.New` without a `context.Context`, we need to load the private key first so we can find our peer ID and add the `PeerIDHook`.